mmengine/runner/runner.py

```diff
@@ -349,19 +346,34 @@ def __init__(
         self._randomness_cfg = randomness
         self.set_randomness(**randomness)
 
-        if experiment_name is not None:
-            self._experiment_name = f'{experiment_name}_{self._timestamp}'
-        elif self.cfg.filename is not None:
+        if experiment_name is None and self.cfg.filename is not None:
             filename_no_ext = osp.splitext(osp.basename(self.cfg.filename))[0]
-            self._experiment_name = f'{filename_no_ext}_{self._timestamp}'
-        else:
-            self._experiment_name = self.timestamp
-        self._log_dir = osp.join(self.work_dir, self.timestamp)
+            experiment_name = f'{filename_no_ext}'
```

Comment:
Is this line necessary? how about:
`experiment_name = osp.splitext(osp.basename(self.cfg.filename))[0]`